plugins/org.eclipse.n4js.ide/src/org/eclipse/n4js/ide/xtext/server/build/ProjectBuilder.java

```diff
@@ -316,9 +317,23 @@ public void doClean(AfterDeleteListener deleteListener, CancelIndicator cancelIn
 			}
 		}
 
+		doCleanIssues();
+
 		doClear();
 	}
 
+	/**
+	 * Send the necessary 'publishDiagnostics' notifications to the LSP client to removing all existing issues located
+	 * in this project. Will not remove those issues from the current project state in {@link #projectStateSnapshot}.
+	 */
+	protected void doCleanIssues() {
```

Comment:
Should this be named `publishCleanIssues` to avoid that somebody accidentally calls this without a following doClear?

Reply:
Good point, it does not make sense to call this new method without doing a doClear() afterwards. I now renamed the method to `doClearWithNotification()` and invoke `doClear()` right from within the new method. See 1c3480f.